Ore/Wood Update

Hornsoup opened the pull request:
Includes:
- Woodworking / Blacksmithing / Leatherworking update to roughly coincide with names for weapons and armor
- Follower update to coincide with these changes
- Update to impact of Strength and Toughness on weaponry, minor multiplier of log10, but players seeing numbers going up is the point.
- Update to accuracy formula to incite a need for Speed stat (about 400 Billion by 9/9/3, doable in testing)
- optional pause on thug, visible/clickable with home settings (thus requires a manual)
- Minor update to recruiting, because only an idiot would use up the last of their money on advertising guaranteed to have no effect.

Hornsoup commented:
Figured out how to reduce that armorer block. Had to use non-null assertions since the compiler was being a butt and not recognizing the logic, that's all.

immortalityidle commented on src/app/game-state/battle.service.ts:
```diff
@@ -208,8 +208,8 @@ export class BattleService {
   }
 
   youAttack(){
+    this.characterService.characterState.accuracy = Math.min((this.troubleKills + Math.sqrt(this.characterService.characterState.attributes.speed.value)) / this.troubleKills / 2, 1)
```
Is this going to fight back and forth with the accuracy setting in the recalculateDerivedStats function?

immortalityidle replied:
Oh. Nevermind. You removed that. Any reason we set this here and not there?

Hornsoup replied:
Because it's easier to use troubleKills from here rather than having to inject battle into character.

immortalityidle commented on src/app/game-state/item-repo.service.ts:
```diff
@@ -625,39 +625,53 @@ export class ItemRepoService {
       value: 9,
       description: 'A spiritual quality log.',
     },
+    redwoodLog: {
```
Redwood is pretty soft. Seems like this should be a lower cost wood. Let's give these unique descriptions too.

Hornsoup replied:
Oop, forgot descriptions. Not sure where to place the woods, because some are useful for structures and others have a way different janka scale than what's already here.

Hornsoup replied:
Kinda want to make elm balsa instead.

Hornsoup replied:
Alright, roughed some locations based on softness since harder woods are more suitable for weapons, existing woods were really wherever they wanted to be.

immortalityidle commented on src/app/game-state/item-repo.service.ts:
```diff
@@ -562,102 +562,116 @@ export class ItemRepoService {
+      value: 9,
       description: 'An amazing quality log.',
     },
-    peachwoodLog: {
```
No objections to the new wood, but can we add peachwood back in there too? That's an actual historical and mythological thing that I liked.

immortalityidle commented on src/app/game-state/item-repo.service.ts:
```diff
@@ -562,102 +562,116 @@ export class ItemRepoService {
       value: 1,
       description: 'A basic animal hide.'
     },
-    elmLog: {
```
Can we keep both?

Hornsoup replied:
I'll drop out redwood for balsa and rubywood for lignum vitae, maybe that's best.

immortalityidle replied:
Sure. Not opposed to adding more wood types if you want though.

kagejohn commented on src/app/game-state/activity.service.ts:
```diff
@@ -1783,7 +1783,7 @@ export class ActivityService {
               if (grade >= 1){ // if the wood was found
                 this.inventoryService.addItem(this.inventoryService.generateWeapon(
-                  (grade * 5) + Math.floor(Math.log2(this.characterService.characterState.attributes.woodLore.value)), 'wood'));
+                  Math.floor(Math.pow(this.characterService.characterState.attributes.metalLore.value, grade/32)), 'wood'));
```
how come it was changed to metal lore for the wood weapon?

Hornsoup replied:
crapity crap that's copypasta

Hornsoup commented:
That should be all the woodworking fixes, anything else I missed?

Hornsoup commented:
Crap, just realized how much impact this would have on current weapons being OP and newer players getting hit hard, and how much impact equipment infusion has.

Hornsoup commented:
Updated a lot of the weapon and armor related balance and moved the gem usage into the weapon and armor generation since it was basically a huge method and added activity that weren't necessary since they did the same thing that automatically using gems was for.
